opensrp-immunization/src/main/java/org/smartregister/immunization/ImmunizationLibrary.java

```diff
@@ -224,9 +224,13 @@ public void setCurrentConditionalVaccine(String currentConditionalVaccine) {
     }
 
     public boolean allowSyncImmediately() {
+        if(instance.getProperties().hasProperty(IMConstants.APP_PROPERTIES.VACCINE_SYNC_IMMEDIATE))
```

Comment:
here we can use the  helper method **instance.getProperties().isTrue(IMConstants.APP_PROPERTIES. VACCINE_SYNC_IMMEDIATE);** which combines both checks into one

Reply:
That will nullify the function `org.smartregister.immunization.ImmunizationLibrary#setAllowSyncImmediately`. I split up the assignment so that the properties value takes precedence but also this allows the method is usable. This is because of flavor build where the properties flavor file is not supported